packages/hub/src/hooks/useMutationForm.ts

```diff
@@ -17,6 +18,9 @@ export function useMutationForm<
   FormShape extends FieldValues = never,
   TMutation extends CommonMutationParameters<TTypename> = never,
   TTypename extends string = never,
+  // In some cases, you might want to pass extra data to the mutation, which isn't available in `formDataToVariables`.
+  // See `SaveButton` implementation in `EditSquiggleSnippetModel` for an example where it's needed.
+  ExtraData extends Record<string, any> = Record<string, never>,
```

Comment:
This part is messy and I wish I had a better idea how to do this, but I couldn't think of anything.

What's going on here:
- `useMutationForm` returns `onSubmit` callback, which is calling `formDataToVariables`
- `formDataToVariables` is allowed to inject extra data to `variables` which are passed to GraphQL query (that's how we handle `version` field, for example, which is currently not part of react-hook-form form)
- but the modal with textarea for comment is in another component! so I couldn't capture it in `formDataToVariables`
- I could make the comment part of the top-level playground form, but I was worried about phantom values (where you would save once with a comment, then save without a comment, but the previous comment would get attached to the mutation because it was never cleaned up)
  - I could clean it up with effects, but it would create a tighter dependency between `<SaveButton>` and the main `<EditSquiggleSnippetModel>` (`SaveButton` and its menu would need to know the details about the form)
  - alternatively, I could pass `setComment` callback to `<SaveButton>`, but that has the same issue
  - so instead I opted for making `useMutationForm` more flexible; it also makes it a bit more messy, with four generic parameter types now, and with a `onSubmit` that can't be passed to `<form onSubmit={onSubmit}>` directly, but I hope that the complexity will pay off in flexibility gains when I run into the similar problem again in the future

Reply:
Yea, this seems messy. I might suggest adding a longer comment there.

It would likely take me a while to come up with something better, so I'll just leave it for now.

Reply:
I put a link to this comment in source.